core/model/Reading.cpp

```diff
@@ -14,45 +14,268 @@
  * limitations under the License.
  */
 
-#include "Reading.h"
+#include "core/model/Reading.h"
 
-#include "core/utilities/StringUtils.h"
+#include <algorithm>
+#include <regex>
+#include <sstream>
+#include <stdexcept>
+#include <utility>
 
 namespace wolkabout
 {
-Reading::Reading(std::vector<std::string> values, std::string reference, unsigned long long int rtc,
-                 std::string deviceKey)
-: m_values(std::move(values)), m_reference(std::move(reference)), m_rtc(rtc), m_key{std::move(deviceKey)}
+Reading::Reading(std::string reference, std::string value, std::uint64_t rtcTimestamp)
+: m_reference(std::move(reference)), m_values({std::move(value)}), m_timestamp(rtcTimestamp)
 {
 }
 
-const std::vector<std::string>& Reading::getValues() const
+Reading::Reading(std::string reference, Location location, std::uint64_t rtcTimestamp)
```

Comment:
I mean, the platform doesn't handle a Location as a two-value feed. It just seems right to handle it the same way, and not bend ourselves trying to make something that in the background will work as a two-value reading, but in fact, it will look on the surface as a single-value feed.